src/java.base/unix/native/libjsig/jsig.c

```diff
@@ -250,6 +260,9 @@ JNIEXPORT int sigaction(int sig, const struct sigaction *act, struct sigaction *
      * within the JVM_begin_signal_setting-JVM_end_signal_setting-window. There can be any number of non-modifying
      * calls, but they will only return the expected preexisting handler if executed before the modifying call.
      */
+    if (deprecated_usage[sig] == true) {
+      print_deprecation_warning();
+    }
```

Comment:
Yes, took me a minute to realise this is needed because back when/if signal was called, we set deprecated_usage[sig] but only called print_deprecation_warning() if jvm_signal_installed && sigused.